psi4/driver/procrouting/sapt/sapt_proc.py

```diff
@@ -34,77 +34,135 @@
 from ...constants import constants
 from ...p4util.exceptions import *
 from .. import proc_util
-from ..proc import scf_helper
+from ..proc import scf_helper, run_scf
 from . import sapt_jk_terms, sapt_mp2_terms, sapt_sf_terms
 from .sapt_util import print_sapt_dft_summary, print_sapt_hf_summary, print_sapt_var
+import qcelemental as qcel
+from ...p4util.exceptions import ConvergenceError
 
 # Only export the run_ scripts
-__all__ = ['run_sapt_dft', 'sapt_dft', 'run_sf_sapt']
+__all__ = ["run_sapt_dft", "sapt_dft", "run_sf_sapt"]
 
 
 def run_sapt_dft(name, **kwargs):
-    optstash = p4util.OptionsState(['SCF_TYPE'], ['SCF', 'REFERENCE'], ['SCF', 'DFT_GRAC_SHIFT'], ['SCF', 'SAVE_JK'])
+    optstash = p4util.OptionsState(
+        ["SCF_TYPE"],
+        ["SCF", "REFERENCE"],
+        ["SCF", "DFT_GRAC_SHIFT"],
+        ["SCF", "SAVE_JK"],
+    )
 
     core.tstart()
     # Alter default algorithm
-    if not core.has_global_option_changed('SCF_TYPE'):
-        core.set_global_option('SCF_TYPE', 'DF')
+    if not core.has_global_option_changed("SCF_TYPE"):
+        core.set_global_option("SCF_TYPE", "DF")
 
     core.prepare_options_for_module("SAPT")
 
-    # Get the molecule of interest
-    ref_wfn = kwargs.get('ref_wfn', None)
-    if ref_wfn is None:
-        sapt_dimer = kwargs.pop('molecule', core.get_active_molecule())
-    else:
-        core.print_out('Warning! SAPT argument "ref_wfn" is only able to use molecule information.')
-        sapt_dimer = ref_wfn.molecule()
-
-    sapt_dimer, monomerA, monomerB = proc_util.prepare_sapt_molecule(sapt_dimer, "dimer")
-
     # Grab overall settings
+    do_mon_grac_shift_A = False
+    do_mon_grac_shift_B = False
     mon_a_shift = core.get_option("SAPT", "SAPT_DFT_GRAC_SHIFT_A")
     mon_b_shift = core.get_option("SAPT", "SAPT_DFT_GRAC_SHIFT_B")
+    SAPT_DFT_GRAC_COMPUTE = core.get_option("SAPT", "SAPT_DFT_GRAC_COMPUTE")
```

Comment:
Having an all-caps variable is a little unusual. And since it's a str option, it should come back reliably all-caps, so I don't think the `.upper()`s are needed.